packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Interact with image and video. It lets the app developer ask the user to get images or video from their camera / camera roll / file system.
+ * @beta
+ */
+export namespace visualMedia {
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * All properties common to Image and Video Props
+   * @beta
+   */
+  interface VisualMediaProps {
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10.
+     */
+    maxVisualMediaCount: number;
+  }
+
+  /**
+   * The required value of the visualMedia files from gallery
+   * @beta
+   */
+  export interface GalleryProps {
+    /**
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Gallery;
+  }
+  /**
+   * The required value of the visualMedia files from camera
+   * @beta
+   */
+  export interface CameraProps {
+    /**
+     * @hidden
+     * The visualMedia source
+     */
+    source: Source.Camera;
+    /**
+     * Optional; indicate if user is allowed to move between front and back camera
+     * Default value is FrontOrRear
+     */
+    cameraRestriction?: CameraRestriction;
+  }
+
+  /**
+   * Indicate if user is allowed to move between front and back camera or stay in front/back camera only
+   * If the camera option requested by the app isn't available, the SDK will silently default to the platform's standard camera.
+   * @beta
+   */
+  export enum CameraRestriction {
```

Comment:
I agree on what @AE-MS mentioned in another thread: Instead of having infinitely growing 'interfaces,' 'constants,' and 'utils,' it would be easier for us to maintain if we keep interfaces closely associated with the functionality to which they are related.